.concourse/kubecf-pool-reconciler.yaml.gomplate

```diff
@@ -0,0 +1,69 @@
+{{ $kind_pool_size := 2 }}
+
+resources:
+- name: kind-environments
+  type: pool
+  source:
+    uri: git@github.com:SUSE/cf-ci-pools.git
+    branch: kubecf-kind-pools
+    pool: kind
+    private_key: ((github-private-key))
+
+- name: catapult
+  type: git
+  source:
+    uri: https://github.com/SUSE/catapult
+  version:
+    ref: 543bc71abc8bd22e5ef81fdf94a7afeb2f99066d
+
+
+deploy_args: &deploy_args
+- -ce
+- |
+  export FORCE_DELETE=true
+  export BACKEND=ekcp
+  export QUIET_OUTPUT=true
+  export CLUSTER_NAME="$CLUSTER_PREFIX-$(date +%s | sha256sum | base64 | head -c 32 ; echo)"
+  CURRENT_CLUSTERS=$(ls kind-environments/kind/**/* | grep -v .keep | wc -l)
+  if [ $CURRENT_CLUSTERS -lt {{ $kind_pool_size }} ]; then
```

Comment:
Instead of the pipeline making this determination, can EKCP do it? Similar to the API for any other infrastructure provider (AKS/ECP/EKS/etc.), if the deploy job tries to provision a resource when the provider is at its limit, the API request to provision said resource will fail.

Reply:
Yes, the EKCP fails if you don't have enough resources. This is a way to just guarantee we don't hit that error and we can always provision with a boundary limit